qa/SidechainTestFramework/scutil.py

```diff
@@ -416,87 +416,82 @@ def is_mainchain_block_included_in_sidechain_block_reference_info(sc_block_refer
 
-        return True
-    except Exception:
-        return False
+            sc_mc_block_previousblockhash = mc_block_json["header"]["hashPrevBlock"]
+            assert_equal(expected_mc_block_previousblockhash, sc_mc_block_previousblockhash)
+
+            is_mac_block_included = True
+        except Exception:
+            continue
+
+    return is_mac_block_included
 
 """
-For given PublicKey25519 verify the number of related regular boxes and verify the sum of their balances.
+Verify the wallet balance is equal to an expected balance.
 
 Parameters:
  - sc_node: a sidechain node
- - array_of_expected_public_keys: array of public keys.
-                    [pubKey_1, pubKey_2, ..., pubKey_n]
- - array_of_expected_boxes_count: array of expected boxes count for each public key of the second parameter.
-                    [boxes_count_for_pubKey_1, boxes_count_for_pubKey_2, ..., boxes_count_for_pubKey_n]
- - array_of_expected_sc_balances: array of expected balances for each public key of the second parameter.
-                    [balance_for_pubKey_1, balance_for_pubKey_2, ..., balance_for_pubKey_n]
+ - expected_wallet_balance
 """
-def check_regularbox_balance(sc_node, array_of_expected_public_keys, array_of_expected_boxes_count, array_of_expected_sc_balances):
+def check_wallet_balance(sc_node, expected_wallet_balance):
+    response = sc_node.wallet_balance()
+    balance = response["result"]
+    assert_equal(expected_wallet_balance * 100000000, int(balance["balance"]), "Unexpected balance")
+
 
-    response = sc_node.wallet_allPublicKeys()
-    public_keys = response["result"]["propositions"]
-    assert_equal(len(array_of_expected_public_keys), len(public_keys), "Unexpected number of public keys")
+"""
+For a given Account verify the number of related regular boxes and verify the sum of their balances.
 
+Parameters:
+ - sc_node: a sidechain node
+ - account: an instance of Account (see sc_bootstrap_info.py)
+ - expected_boxes_count: the number of expected boxes for that account
+ - expected_balance: expected balance for that account
+"""
+def check_regularbox_balance(sc_node, account, expected_boxes_count, expected_balance):
     response = sc_node.wallet_allBoxes()
     boxes = response["result"]["boxes"]
+    boxes_balance = 0
+    boxes_count = 0
+    pub_key = account.publicKey
+    for box in boxes:
+        if box["proposition"]["publicKey"] == pub_key and box["typeId"] == 1:
+            box_value = box["value"]
+            assert_true(box_value > 0,
+                        "Non positive value for box: {0} with public key: {1}".format(box["id"], pub_key))
+            boxes_balance += box_value
+            boxes_count += 1
+
+    assert_equal(expected_boxes_count, boxes_count,
+                "Unexpected number of boxes for public key {0}. Expected {1} but found {2}."
+                .format(pub_key, expected_boxes_count, boxes_count))
+    assert_equal(expected_balance * 100000000, boxes_balance,
```

Comment:
FYI: Later we should think about coins presentation in MC and SC. In MC they are displayed in Zen, in SC - satoshi.